extensions/typescript-language-features/src/typeScriptServiceClientHost.ts

```diff
@@ -150,6 +151,60 @@ export default class TypeScriptServiceClientHost extends Disposable {
 					standardFileExtensions: [],
 				}, onCompletionAccepted);
 			}
+			// TODO: Still probably isn't the right place for this
+			const ext = vscode.extensions.getExtension('github.copilot');
+			if (!ext) {
+				vscode.window.showErrorMessage(vscode.l10n.t('Could not find related-files extension'));
+				return;
+			}
+			await ext.activate();
+			const relatedAPI = ext.exports as {
+				registerRelatedFilesProvider(
+					providerId: { extensionId: string; languageId: string },
+					callback: (uri: vscode.Uri) => Promise<{ entries: vscode.Uri[]; traits?: { name: string; value: string }[] }>
+				): void;
+			} | undefined;
+			if (relatedAPI) {
```

Comment:
Does this also need to be gated on the TS server version?

Reply:
I expect it to ship in 5.7. Gating it is a good idea. How do I do that? I saw a lot of `// @ts-expect-error until ts 5.6` in the code base, accompanied by `if (event.body.entries)` kinds of checks. Is that the right way?

Reply:
Ah, the language features are gated on API.v560 (for example). Is it OK to add API.v570?